dotCMS/src/integration-test/java/com/dotcms/storage/StoragePersistenceAPITest.java

```diff
@@ -0,0 +1,269 @@
+                new TestCase(StorageType.FILE_SYSTEM, groupName),
+                new TestCase(StorageType.DB, groupName)
+        };
+    }
+
+    /**
+     * Test Scenario: Regardless of group we attempt saving a binary
+     * Expected Results: TBD
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("getDupeBinaryTestCases")
+    public void Test_Attempt_Push_Same_Binary_Twice(final TestCase testCase) {
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        storage.deleteObject(testCase.groupName, testCase.path);
+        assertTrue(storage.createGroup(testCase.groupName));
+        final Object object = storage
+                .pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+        assertNotNull(object);
+
+        assertTrue(storage.existsObject(testCase.groupName, testCase.path));
+
+        final Object object2 = storage
+                .pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+        assertNotNull(object2);
+    }
+
+    @DataProvider
+    public static Object[] getDupeBinaryTestCases() throws Exception{
+        final String groupName = RandomStringUtils.randomAlphanumeric(10);
+
+        final File imageFile = new File(
+                ConfigTestHelper.getUrlToTestResource(TEST_IMAGE_PNG).toURI());
+
+        return new Object[]{
+
+                new TestCase(StorageType.FILE_SYSTEM,
+                        groupName, imageFilePath, imageFile
+                ),
+
+                new TestCase(StorageType.DB,
+                        groupName, imageFilePath, imageFile
+                )
+
+        };
+    }
+
+    /**
+     * Given scenario: We want corroborate that we're able to recover files and verify the existence of a group regardless of casing.
+     * Expected Results: Regardless of casing we should be able to retrieve and find objects
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("getRandomTestCases")
+    public void Test_Pull_File_Different_Casing(final TestCase testCase) {
+        final StoragePersistenceAPI storage = persistenceProvider.getStorage(testCase.storageType);
+        if(!storage.existsGroup(testCase.groupName)){
+           assertTrue(storage.createGroup(testCase.groupName));
+        }
+        final Object object = storage
+                .pushFile(testCase.groupName, testCase.path, testCase.file, ImmutableMap.of());
+        assertNotNull(object);
+
+        final String randomCaseGroupName = upperCaseRandom(testCase.groupName, testCase.groupName.length());
+        assertTrue(storage.existsGroup(randomCaseGroupName));
+        final String randomCasePath = upperCaseRandom(testCase.path, testCase.path.length());
+        final File file = storage.pullFile(randomCaseGroupName, randomCasePath);
+        assertNotNull(file);
+
+    }
+
+    @DataProvider
+    public static Object[] getRandomTestCases() throws Exception{
```

Comment:
Codacy found an issue: [A method/constructor should not explicitly throw java.lang.Exception](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)